core/src/main/java/org/opensearch/sql/planner/physical/UnnestOperator.java

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.planner.physical;
+
+import static java.util.stream.Collectors.mapping;
+import static java.util.stream.Collectors.toList;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import org.apache.commons.lang3.StringUtils;
+import org.opensearch.sql.data.model.ExprCollectionValue;
+import org.opensearch.sql.data.model.ExprTupleValue;
+import org.opensearch.sql.data.model.ExprValue;
+import org.opensearch.sql.expression.ReferenceExpression;
+
+@EqualsAndHashCode(callSuper = false)
+public class UnnestOperator extends PhysicalPlan {
+  @Getter
+  private final PhysicalPlan input;
+  @Getter
+  private final Set<String> fields; // Needs to be a Set to match legacy implementation
+  @Getter
+  private final Map<String, List<String>> groupedPathsAndFields;
```

Comment:
Do you need to store both? You can calculate second field from first when you need.
Use `LinkedHashMap` to keep the insert order.

Reply:
I think this is the easiest way to accomplish the comparison in `containSamePath` function. When adding a new key to the result map we need to check: 

- Is there any matching keys with the same path.
- Is a matching key with the same path already in the result map.

Rather than calculating the path for each field already in the results map, and each field being added to the results map we maintain this member variable.